nengo/connection.py

```diff
@@ -374,10 +374,16 @@ def _str(self):
         if self.label is not None:
             return self.label
 
+        desc = None
+        if self.function is not None and hasattr(self.function, "__name__"):
+            desc = self.function.__name__
+        elif not hasattr(self.function, "__name__"):
+            desc = str(self.function)
```

Comment:
You can do this in one line with `desc = getattr(self.function, '__name__', str(self.function)) if self.function is not None else ""`, and then you don't have to check `desc is not None` below either.